testing/trino-testing-containers/src/main/java/io/trino/testing/ceph/CephClient.java

```diff
@@ -0,0 +1,142 @@
+
+import static java.util.regex.Matcher.quoteReplacement;
+
+public class CephClient
+{
+    private static final Logger log = Logger.get(CephClient.class);
+
+    private static final Set<String> createdBuckets = Sets.newConcurrentHashSet();
+
+    private final AmazonS3 client;
+
+    public CephClient(String endpoint, String accessKey, String secretKey)
+    {
+        AWSCredentials credentials = new BasicAWSCredentials(accessKey, secretKey);
+        AWSCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(credentials);
+        ClientConfiguration clientConfiguration = new ClientConfiguration()
+                .withProtocol(Protocol.HTTP)
+                .withSignerOverride("AWSS3V4SignerType");
+
+        client = AmazonS3ClientBuilder
+                .standard()
+                .withCredentials(credentialsProvider)
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endpoint, Regions.US_EAST_1.name()))
+                .withPathStyleAccessEnabled(true)
+                .withClientConfiguration(clientConfiguration)
+                .build();
+    }
+
+    public void copyResourcePath(String bucket, String resourcePath, String target)
+    {
+        ensureBucketExists(bucket);
+
+        try {
+            ClassPath.from(CephClient.class.getClassLoader())
+                    .getResources().stream()
+                    .filter(resourceInfo -> resourceInfo.getResourceName().startsWith(resourcePath))
+                    .forEach(resourceInfo -> {
+                        String fileName = resourceInfo.getResourceName().replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(target));
+                        try {
+                            putObject(bucket, fileName, resourceInfo.asByteSource());
+                        }
+                        catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+        }
+        catch (IOException e) {
+            log.warn(e, "Could not copy resources from classpath");
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void putObject(String bucket, String fileName, ByteSource byteSource)
+            throws IOException
+    {
+        ensureBucketExists(bucket);
+
+        try (InputStream inputStream = byteSource.openStream()) {
+            client.putObject(new PutObjectRequest(bucket, fileName, inputStream, new ObjectMetadata()));
+        }
+    }
+
+    public List<String> listObjects(String bucket, String path)
+    {
+        return Streams
+                .stream(S3Objects.withPrefix(client, bucket, path))
+                .map(S3ObjectSummary::getKey).collect(Collectors.toList());
+    }
+
+    public void makeBucket(String bucket)
+    {
+        if (!createdBuckets.add(bucket)) {
+            // Forbid to create a bucket with given name more than once per class loader.
```

Comment:
per this message, `io.trino.testing.minio.MinioClient#createdBuckets` and `io.trino.testing.ceph.CephClient#createdBuckets` must be disjoint. we should ensure that

cc @losipiuk